dandi/download.py

```diff
@@ -313,10 +313,23 @@ def download_generator(
                     lock=lock,
                 )
 
+            def _guarded_generator(path, generator):
```

Comment:
There's no need to nest this function inside `download_generator()`.  It would be better placed at module level.

Reply:
I'm not aware of any recommendations against this sort of thing; I just don't like nesting functions for no reason.  I'll leave it up to you.

Reply:
followed your recommendation